modules/ROOT/pages/migration.adoc

```diff
@@ -0,0 +1,937 @@
+var config = new ReplicatorConfiguration(database);
+
+// After
+var collections = CollectionConfiguration.FromCollections(DefaultCollection);
+var config = new ReplicatorConfiguration(collections, endpoint);
+
+// Clone immutably
+var updated = config with { ReplicatorType = ReplicatorType.Push };
+----
+
+== Replication Pending Documents
+
+The replication pending documents API applies only to documents that are waiting to be pushed to the remote endpoint.
+The deprecated methods for retrieving or checking pending documents without specifying a collection have been removed in 4.0.0.
+Use the corresponding methods that require a Collection object instead.
+
+[{tabs}]
+====
+Swift::
++
+--
+[source,swift]
+----
+// Before
+let pendings = try replicator.pendingDocumentIds()
+let isPending = try replicator.isDocumentPending("doc1")
+
+// After
+let defaultCollection = try database.defaultCollection()
+let pendings = try replicator.pendingDocumentIds(collection: defaultCollection)
+let isPending = try replicator.isDocumentPending("doc1", collection: defaultCollection)
+----
+--
+
+Objective-C::
++
+--
+[source,objectivec]
+----
+// Before
+NSArray* pendings = [replicator pendingDocumentIds: &error];
+BOOL isPending = [replicator isDocumentPending: @"doc1" error: &error];
+
+// After
+CBLCollection* defaultCollection = [database defaultCollection: &error];
+NSArray* pendings = [replicator pendingDocumentIDsForCollection: defaultCollection
+                                                           error: &error];
+BOOL isPending = [replicator isDocumentPending: @"doc1"
+                                    collection: defaultCollection
+                                         error: &error];
+----
+--
+
+Java::
++
+--
+[source,java]
+----
+// Before
+Set<String> pendings = replicator.getPendingDocumentIds();
+boolean isPending = replicator.isDocumentPending("doc1");
+
+// After
+Collection defaultCollection = database.getDefaultCollection();
+Set<String> pendings = replicator.getPendingDocumentIds(defaultCollection);
+boolean isPending = replicator.isDocumentPending("doc1", defaultCollection);
+----
+--
+
+C#::
++
+--
+[source,csharp]
+----
+// Before
+var pendings = replicator.GetPendingDocumentIds();
+var isPending = replicator.IsDocumentPending("doc1");
+
+// After
+var defaultCollection = database.GetDefaultCollection();
+var pendings = replicator.GetPendingDocumentIds(defaultCollection);
+var isPending = replicator.IsDocumentPending("doc1", defaultCollection);
+----
+--
+
+C::
++
+--
+[source,c]
+----
+// Before
+FLDict pendings = CBLReplicator_PendingDocumentIDs(replicator, &error);
+bool isPending = CBLReplicator_IsDocumentPending(replicator, "doc1", &error);
+
+// After
+CBLCollection* defaultCollection = CBLDatabase_DefaultCollection(database, &error);
+FLDict pendings = CBLReplicator_PendingDocumentIDs2(replicator, defaultCollection, &error);
+bool isPending = CBLReplicator_IsDocumentPending2(replicator, "doc1", defaultCollection, &error);
```

Comment:
There should be no `2` here.
CBLReplicator_PendingDocumentIDs(replicator, defaultCollection, &error);
CBLReplicator_IsDocumentPending(replicator, "doc1", defaultCollection, &error);

These have been renamed in 4.0 when we removed the old ones. Technically the API name stayed the same, implementation was moved over

Reply:
I've updated this in a commit.